pkg/pool-manager/pool_test.go

```diff
@@ -375,6 +378,66 @@ var _ = Describe("Pool", func() {
 				Expect(exist).To(BeFalse())
 			})
 		})
+		Context("check create a vm with mac address allocation", func() {
+			var (
+				newVM        *kubevirt.VirtualMachine
+				poolManager  *PoolManager
+				allocatedMac string
+			)
+			BeforeEach(func() {
+				poolManager = createPoolManager("02:00:00:00:00:00", "02:00:00:00:00:01", &vmConfigMap)
+				newVM = masqueradeVM.DeepCopy()
+				newVM.Name = "newVM"
+
+				By("Create a VM")
+				err := poolManager.AllocateVirtualMachineMac(newVM)
+				Expect(err).ToNot(HaveOccurred(), "should successfully  allocated macs")
+
+				allocatedMac = newVM.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress
+				By("check mac allocated as expected")
+				Expect(allocatedMac).To(Equal("02:00:00:00:00:01"), "should successfully allocate the first mac in the range")
```

Comment:
Move to `It`

Reply:
do you mean in separate `It`?

Reply:
We can check this at an `It("should successfully allocate the first mac in the range")` after this `BeforeEach`